File: kora-app-annotation-processor/src/test/java/ru/tinkoff/kora/kora/app/annotation/processor/app/AppWithFactories6.java
package ru.tinkoff.kora.kora.app.annotation.processor.app;

import ru.tinkoff.kora.common.KoraApp;
import ru.tinkoff.kora.common.annotation.Root;

@KoraApp
public interface AppWithFactories6 {

    default <T> GenericClass<T> factory1(GenericClass<T> genericClass) {
        return new GenericClass<>();
    }

    @Root
    default Class2 class2(GenericClass<Class1> class1) {
        return new Class2();
    }


    final class GenericClass<T> {}

    class Class1 {}

    class Class2 {}
}

File: kora-app-symbol-processor/src/test/kotlin/ru/tinkoff/kora/kora/app/ksp/app/AppWithFactories6.kt
package ru.tinkoff.kora.kora.app.ksp.app

import ru.tinkoff.kora.common.KoraApp
import ru.tinkoff.kora.common.annotation.Root

@KoraApp
interface AppWithFactories6 {
    fun <T> factory1(genericClass: GenericClass<T>): GenericClass<T> {
        return GenericClass()
    }

    @Root
    fun class2(class1: GenericClass<Class1>): Class2 {
        return Class2()
    }

    class GenericClass<T>
    class Class1
    class Class2
}
